src/AppBundle/Service/OrderManager.php

```diff
@@ -186,20 +190,19 @@ public function capturePayment(OrderInterface $order)
         $stateMachine = $this->stateMachineFactory->get($stripePayment, PaymentTransitions::GRAPH);
 
         $stripeAccount = $order->getRestaurant()->getStripeAccount();
+        $restaurantPaysStripeFee = $order->getRestaurant()->getContract()->isRestaurantPaysStripeFee();
+        $stripeOptions = array();
 
-        if (!is_null($stripeAccount)) {
-            $restaurantStripeId = $stripeAccount->getStripeUserId();
-        } else {
-            $restaurantStripeId = null;
+        if (!is_null($stripeAccount) && $restaurantPaysStripeFee) {
+            $stripeOptions['stripe_account'] = $stripeAccount->getStripeUserId();
         }
 
         try {
 
             $charge = Stripe\Charge::retrieve(
                 $stripePayment->getCharge(),
-                array(
-                    'stripe_account' => $restaurantStripeId
-            ));
+                $stripeOptions
+            );
```

Comment:
I think there is something missing after that: we still need to create a Transfer to the Stripe account of the restaurant. See faf97fa5d5a52ba9c9188f9ea526e3c2b58e43ea

Reply:
There is no need to create a transfer when using destination charges. Before https://github.com/coopcycle/coopcycle-web/commit/faf97fa5d5a52ba9c9188f9ea526e3c2b58e43ea we were using "separate charges and transfers": see https://stripe.com/docs/connect/charges